### PR TITLE
UPSTREAM: <drop>: Double the global timeout if performing a global list

### PR DESCRIPTION
**Pull request by smarterclayton**
In 3.6, users still on etcd may be seeing large list calls take in
excess of 1 minute (the hard limit). In 3.7, all users will be on etcd3
and protobuf and we expect times to drop by an order of magnitude.
Since our largest in the wild clusters are taking about 1 minute 3
seconds, double the global timeout limit for global lists only for 3.6,
and in 3.7 this will no longer be necessary.

@deads2k

### CODE REVIEW

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go [hunk 1]
```diff
@@ -56,6 +56,13 @@ func WithTimeoutForNonLongRunningRequests(handler http.Handler, requestContextMa
 		if longRunning(req, requestInfo) {
 			return nil, nil
 		}
+
+		// when performing global calls, increase the timeout for very large clusters
+		timeout := globalTimeout
+		if requestInfo.IsResourceRequest && requestInfo && len(requestInfo.Namespace) == 0 {
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go]: intent is good, but this doesn't compile

**Comment by smarterclayton** [comment]
@liggitt

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go [hunk 2]
```diff
@@ -56,6 +56,13 @@ func WithTimeoutForNonLongRunningRequests(handler http.Handler, requestContextMa
...
 			return nil, nil
 		}
+
+		// when performing global calls, increase the timeout for very large clusters
+		timeout := globalTimeout
+		if requestInfo.IsResourceRequest && len(requestInfo.Namespace) == 0 {
+			timeout = timeout * 2
+		}
+
 		return time.After(globalTimeout), apierrors.NewServerTimeout(schema.GroupResource{Group: requestInfo.APIGroup, Resource: requestInfo.Resource}, requestInfo.Verb, 0)
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go]: Use the local var here, right?

**Comment by smarterclayton** [comment]
journald log snip

/retest

**Comment by smarterclayton** [comment]
/retest
Flake in TestProjectRequest

**Pull request merged**